sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TraceHelperTests.cs

```diff
@@ -236,11 +236,43 @@ public void ActivityWithExceptionEventCreatesExceptionTelemetry()
             var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(batch, "roleName", "roleInstance", "00000000-0000-0000-0000-000000000000");
 
             Assert.Equal(2, telemetryItems.Count());
-            Assert.Equal("Exception", (IEnumerable<char>)telemetryItems[0].Name);
-            Assert.Equal("Request", (IEnumerable<char>)telemetryItems[1].Name);
-            Assert.Equal(exceptionMessage, (IEnumerable<char>)(telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().Message);
-            Assert.Equal("System.Exception", (IEnumerable<char>)(telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().TypeName);
-            Assert.Equal("System.Exception: Exception Message", (IEnumerable<char>)(telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().Stack);
+            Assert.Equal("Exception", telemetryItems[0].Name);
+            Assert.Equal("Request", telemetryItems[1].Name);
+            Assert.Equal(exceptionMessage, (telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().Message);
+            Assert.Equal("System.Exception", (telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().TypeName);
+            Assert.Equal("System.Exception: Exception Message", (telemetryItems[0].Data.BaseData as TelemetryExceptionData).Exceptions.First().Stack);
+        }
+
+        [Fact]
+        public void ActivityWithEventCreatesTraceTelemetry()
+        {
+            var eventName = "Custom Event";
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+               ActivityName,
+               ActivityKind.Server);
+
+            var tagsCollection = new ActivityTagsCollection
+            {
+                { "key1", "value1" },
+            };
+
+            var activityEvent = new ActivityEvent(eventName, default, tagsCollection);
+
+            activity.AddEvent(activityEvent);
+
+            Activity[] activityList = new Activity[1];
+            activityList[0] = activity;
+            Batch<Activity> batch = new Batch<Activity>(activityList, 1);
+
+            var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(batch, "roleName", "roleInstance", "00000000-0000-0000-0000-000000000000");
+
+            Assert.Equal(2, telemetryItems.Count());
+            Assert.Equal("Message", telemetryItems[0].Name);
+            Assert.Equal("Request", telemetryItems[1].Name);
+            Assert.Equal(eventName, (telemetryItems[0].Data.BaseData as MessageData).Message);
```

Comment:
TraceTelemetry has other fields, which should be validated too?
https://learn.microsoft.com/en-us/azure/azure-monitor/app/data-model-trace-telemetry#severity-level

Reply:
it will be null. Added that as well.

Reply:
What does null mean here? Will ingestion covert nulls to something?

Reply:
it will be converted to 0.